### ISSUE TITLE
add Observer Aborts functionality

### ISSUE THREAD

**Issue by GSGBen** [GSGBen/BTUtilityPlugin#2]
- [x] examine existing engine code
- [x] compare engine code to plugin code
- [ ] implement
- [ ] create a pull request

**Comment by GSGBen** [comment]
Tried to get it working with per-tick aborts like the engine BTDecorator_BlueprintBase. Can't get it to tick for the life of me, but only when a decorator is on the Utility selector. It works fine on other selectors. Going to give it a rest and work on getting the blackboard-listener one working, which should be more performant anyway.

**Comment by GSGBen** [comment]
Ah it might be the architecture of the behaviour tree itself: was playing with the blackboard-watching version and it was only changing after the delay too. Logged BecomeRelevant and CeaseRelevant and that only happens when the node it's attached to is being executed. So when i had the working test with the selector before, the nodes I attached were always on the first which was always running so it could tick, become relevant etc. Just tested with the blackboard one on a sequence and it goes through to the wait then has the same issue. Tested with a selector and it worked when it was on the first node, but had the same issue on the second.

The path is always on the utility node though, so could do something where it is reevaluating the priority list each x interval, and call the abort or RequestExecute(this). Probably only good for small utility designs though. Could also have it watching the blackboard entries its children are using. Could do it automatically with some interation and casting.
